Feature/add minting to transfer 726

0xvangrim opened the pull request:
# Description
Issue: https://github.com/DXgovernance/dxvote/issues/726

- Enabled REP minting for REP guilds
- Calling the contract `ERC20SnapshotRep` mint-function

TODO: 
- Add integration tests.
- Fix dev-script to work with REP guilds

`Modal`
![Screenshot 2022-04-13 at 23 20 05](https://user-images.githubusercontent.com/59825171/163273517-c1168107-4201-492a-a23c-b22a86630e5e.png)

`RepMintEditor`
![Screenshot 2022-04-13 at 23 20 26](https://user-images.githubusercontent.com/59825171/163273548-cc6d8fbb-49a3-4637-9136-9789d77e0181.png)

`RepMintInfoLine`
![Screenshot 2022-04-13 at 23 20 33](https://user-images.githubusercontent.com/59825171/163273558-98d805eb-70bc-47c3-9f03-26a88501ce42.png)

`RepMintSummary`
![Screenshot 2022-04-13 at 23 20 45](https://user-images.githubusercontent.com/59825171/163273571-356a3d80-df9c-4850-a811-643a5b6578dd.png)


## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

This has only been manually tested. Since the dev-script is currently not working with REP guilds, it was not possible to execute the actual action. Thus, it needs to be tested that it actually sends the correct data in the calldata.

# Checklist:

- [X ] My code follows the existing style of this project
- [X ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ x ] Any UI changes have been tested and made responsive for mobile views

Commit:
Merge branch 'develop' of github.com:DXgovernance/dxvote

Commit:
Merge branch 'develop' of github.com:DXgovernance/dxvote

Commit:
Merge branch 'develop' of github.com:DXgovernance/dxvote

Commit:
feat: started implementing REP mint functionality.UI is done

Commit:
fix: shows the users address when minting by default now

Commit:
feat: enabled full functionality according to specs. Last part is to sync with a contract

Commit:
feat: mint function now reads from the call

Commit:
feat: added REPGuild check, dynamic token name, and Numerical input

hhamud commented on src/components/Guilds/ActionsBuilder/SupportedActions/REPMint/REPMintSummary.tsx:
```diff
@@ -0,0 +1,64 @@
+  const { data } = useGuildConfig(guildId);
+  const { data: tokenData } = useERC20Info(data?.token);
+  console.log({ tokenData });
+  console.log(tokenData.symbol);
```
console.log removal

MiltonTulli replied:
Can we remove this logs?

hhamud commented on src/components/Guilds/ActionsBuilder/SupportedActions/REPMint/REPMintEditor.tsx:
```diff
@@ -0,0 +1,150 @@
+    };
+  }, [decodedCall]);
+
+  const { imageUrl } = useENSAvatar(parsedData?.toAddress, MAINNET_ID);
```
This chunk of code is very similar throughout the REPMint components. I suggest that it would be worth placing this processing into its own hook.

MiltonTulli commented on src/components/Guilds/ActionsBuilder/SupportedActions/REPMint/REPMintEditor.tsx:
```diff
@@ -0,0 +1,150 @@
+    }
+  };
+  return (
+    <div>
```
Could this be a fragment?

Commit:
chore: removed logs and added hooks

Commit:
chore: format

Commit:
chore: removed code

Commit:
chore: replace div with Fragment

0xvangrim commented:
Hello guys @MiltonTulli @hhamud, back from the dead here in ETHAmsterdam :D. Thanks for the comments, the latest commits do:

- Added two reusable hooks `useTokenData` and `useTotalSupply`
-  Remove `console.log`
- Replaced `div` with `React.Fragment`
